doc/developer/platform/ux.md

```diff
@@ -1,20 +1,902 @@
+account. All API access should be associated with a user within the account.
+The existing "Workspace API tokens" feature in Frontegg should be disabled.
+
+### User
+
+A user is an entity that accesses Materialize Cloud. That entity may be an
+actual human, or a service account for a machine.
+
+A user is uniquely identified by an email address. A user can never change her
+email address. To effectively change her email address, she must create a new
+account with a new email address and delete the old account. This limitation is
+not desirable but a result of a design decision in our authentication provider.
+The upside is that it is safe to use an email address as a stable and unique
+identifier throughout the product. (TODO: verify with Frontegg.)
+
+A user has one password and any number of personal API tokens. A personal API
+token consists of a UUID client ID and a UUID secret.
+
+There are two special account-level roles that a user can belong to: "account
+owner" and "account member." Account owners have access to the global API (i.e.,
+access to billing and enabling new regions) and the ability to invite new users
+to the account, while account members do not. The first user in an account is
+always an account owner. Account owners are effectively superusers of the region
+API, and are not subject to any [per-region roles](#role) when accessing the
+region API.
+
+### App-specific password
+
+An *app-specific password* is an alternate encoding of a personal API token
+for authenticating with the PostgreSQL protocol. Users should create an
+app-specific password for each external service that will connect to Materialize
+Platform via the PostgreSQL protocol.
+
+Rationale: the PostgreSQL protocol does not have separate fields for an API
+client ID and API secret key, so both values need to be packed into the password
+field. Naively packing the UUIDs is unwieldy, however:
+
+```
+893677c2-a74c-4ecb-804d-3b7fa0c61fdc:68104da2-3417-442e-a614-f078dac470be
+```
+
+But since UUIDs are just 16 bytes each, we can pack them into a much shorter
+base64-encoded string:
+
+```py
+>>> base64.urlsafe_b64encode(client_id.bytes + secret_key.bytes).decode().rstrip('=')
+iTZ3wqdMTsuATTt_oMYf3GgQTaI0F0QuphTweNrEcL4
+```
+
+When encoded in this form, we call the value an app-specific password rather
+than an API key.
+
+In the web UI, we will need to build a custom interface for managing personal
+API tokens that shows the app-specific password form alongside the client ID
+and secret.
+
+### Billing
+
+Materialize Platform will bill monthly based on usage. Accounts will be billed
+in terms of compute units, storage units, and network transfer units. A strawman
+proposal: compute units are virtual machine-hours, storage units are
+gigabyte-months, and network transfer units are gigabytes. Only outbound
+network traffic is billed.
+
+We expect metering the compute usage of clusters, the storage usage of sources
```

Comment:
also the compute usage of storage - a one time cost for the compute we spin up in the storage side of the house. in practice, this will result in a pricing model of something like "10c/gigabyte ingested + 10c/gigabyte-month" or something of that sort (i made up those numbers).

Reply:
👍🏽 added to the doc.